### PR TITLE
Enumerables test

### PR DESCRIPTION
**Pull request by uzairali19**
# This is a feature branch of the enumerable-test development branch.

# Advanced Building Blocks - Enumerables

The project is based on reproducing the enumerable methods to their usable states. The functions would have the prefix `#my_` attached to them, eg: the `#each` function would be `#my_each`.

Functions to test :

- `#my_each`
- `#my_each_with_index`
- `#my_select`
- `#my_all?`
- `#my_any?`
- `#my_none?`
- `#my_count`
- `#my_map`
- `#my_inject`

## Authors

👤 **Rockebell**

- Twitter: [@twitterhandle](https://twitter.com/AnassTantane)
- LinkedIn: [LinkedIn](https://www.linkedin.com/in/anass-tantane)
- GitHub: [@githubhandle](https://github.com/rocker-bell)

👤 **Uzair**

- LinkedIn: [LinkedIn](https://www.linkedin.com/in/uzair-ali-964187166/)
- GitHub: [@githubhandle](https://github.com/uzairali19)

### CODE REVIEW

**Review by Elbie-em — COMMENTED** [review]
## STATUS: INVALID 🔴 

Hello @uzairali19, great effort making it this far with your project, however,  I will have to mark this review request as invalid because you did not set up your **Linter**.
Screenshot 2021-05-20 at 12 01 11
This is a mandatory requirement, Linters help make your code clean and professional.

Kindly apply the changes and request another code review.

*All the best, you are almost there* 💪

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          result = false
+        end
+      end
+    else
+      "Please input a block"
+    end
+
+    puts result
+  end
+
```
Comment (Elbie-em) [inline enumerables.rb:55]: ### my_all

- [ ] my_all should always return true if the block given never returns false or nil

- [ ] this method should return false if the block given is false or nil

- [ ] when no block or argument is given my_all should return false when one of the collection members is false or nil

- [ ] When a class is passed as an argument, it should return true if all members of that collection are of that class e.g Integer and vice-versa(return false) if members of that collection are all not of that class.

- When a Regex is passed as an argument, it returns false if any of the collection does not match the Regex and vice-versa(returns true) if there is a match.

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        else
+          next
+        end
+      end
+    else
+      "Please input a block"
+    end
+    puts result
+  end
+
```
Comment (Elbie-em) [inline enumerables.rb:70]: ### my_any 

- [ ]  This method should return false if the block ever returns a value false or nil

- [ ] When no block or argument is given it should return true if at least one of the collection is not false or nil and vice-versa(returns false) if at least one of the collection is not true.

- [ ] When a class is passed as an argument, it should return true if one member of that collection are of that class e.g Integer

- [ ]  When a Regex is passed as an argument, it returns true if any of the collection matches the Regex.

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        else
+          result = true
+        end
+      end
+    else
+      "Please input a block"
+    end
+    puts result
+  end
+
```
Comment (Elbie-em) [inline enumerables.rb:87]: ### my_none 

- [ ]  This method returns true if the block given never returns true for all elements and vice-versa(returns false) if the block ever returns true for all elements.

- [ ]  When a class is passed as an argument, it should return true if none of the members of that collection are of that class e.g Integer and vice-versa(return false)  if any or all the members of that collection are of that class.

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+          count += 1
+        end
+      end
+    else
+      count = self.length
+    end
+
+    return count
+  end
+
```
Comment (Elbie-em) [inline enumerables.rb:110]: ### my_count 

- [ ] This method should return the number of items in the enum given.

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      for i in 0..self.length - 1
+        result.push(yield(self[i]))
+      end
+    else
+      result = self
+    end
+
+    return result
+  end
+
```
Comment (Elbie-em) [inline enumerables.rb:123]: ### my_map 

- [ ]  This method should return an Enumerator when no block is given

- [ ] This method should return a new array with the results of running a given block once for every element an enum.

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+          result = result.send(startVal, element)
+        end
+        return result
+      else
+        return "You didn't provide the correct parameters"
+      end
+    end
+  end
+end
+
```
Comment (Elbie-em) [inline enumerables.rb:166]: ### my_inject

- [ ]  This method raises a LocalJump error when no block is given.

- [ ]  When a block is given with an initial value it should combine all the elements of enum by applying a binary operation, specified by a block

- [ ]  When a symbol is specified without an initial value it should combine each element of the collection by applying the symbol as a named method.

**Review by Elbie-em — CHANGES_REQUESTED** [review]
## STATUS CHANGES REQUIRED: 🟡  ♻️ 

Hi @uzairali19, good job configuring your linter, however in order for this project to be approved you need to make some changes.

###  - [ ]  Please check the comments below ⬇️ 

*Feel free to use the test cases in these [docs](https://ruby-doc.org/core-3.0.1/Enumerable.html). By comparing the methods you create to the original and using the test cases, you will be able to get accurate results.*

### *As soon as you are done, request another code review*

*You are almost there,All the best*

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Review by umairarshadbutt — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.
- Congratulations on the splendid job @uzairali19 and @rocker-bell ! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Pull request merged**